### PR TITLE
[CoSim] data-transfer check

### PR DESCRIPTION
**Pull request by philbucher**
This PR improves the `CoSimulationDataTransferOperator`
basically the checks are better integrated

### CODE REVIEW

**Comment by philbucher** [comment]
Ping @adityaghantasala

File: applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_data_transfer_operator.py [hunk 1]
```diff
@@ -11,14 +11,32 @@ def __init__(self, settings):
         self.settings = settings
         self.settings.ValidateAndAssignDefaults(self._GetDefaultSettings())
         self.echo_level = self.settings["echo_level"].GetInt()
+        self.__checked_combinations = []
 
     def TransferData(self, from_solver_data, to_solver_data, transfer_options):
-        raise NotImplementedError("This function has to be implemented in the derived class!")
+        # 1. Check if specified transfer options are available
```
Comment (adityaghantasala) [inline applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_data_transfer_operator.py]:
```suggestion
        # 1. Check if specified transfer options are available. 
        #     IMPORTANT: To be implemented by derived classes
```

File: applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_data_transfer_operator.py [hunk 2]
```diff
@@ -11,14 +11,32 @@ def __init__(self, settings):
...
+        self._CheckAvailabilityTransferOptions(transfer_options)
```
Comment (adityaghantasala) [inline applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_data_transfer_operator.py:18]: This function _CheckAvailability is to be implemented by the derived classes or ?

Comment (philbucher) [reply applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_data_transfer_operator.py:18]: https://github.com/KratosMultiphysics/Kratos/blob/64a93514f881b2cf1351baedba03306d0fab1134/applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_data_transfer_operator.py#L28-L30

**Review by adityaghantasala — APPROVED** [review]

**Comment by philbucher** [comment]
thanks

**Pull request merged**